NH-106175 Python: Maintain SW_APM_TRANSACTION_NAME support in pure Python / OTLP-default

jerrytfleung opened the pull request:
- Removed `calculate_otlp_transaction_name`
- Considered `SW_APM_TRANSACTION_NAME` in serviceentry_processor
- Retrieved txn name from span in SolarWindsOTLPMetricsSpanProcessor
- Updated unit test cases

https://swicloud.atlassian.net/browse/NH-106175

tammy-baylis-swi commented on solarwinds_apm/trace/serviceentry_processor.py:
```diff
@@ -78,12 +78,15 @@ def on_start(
         # or serverless name. Otherwise, use the span's name
         pool = get_transaction_name_pool()
+
+        sw_apm_txn_name = os.environ.get("SW_APM_TRANSACTION_NAME", None)
```
Since is "the" place that default transaction naming happens now, please could we update the docstring (L68) to something like this:

```
        """Calculates default transaction name for span and metrics following this order
        of decreasing precedence, truncated to 255 char:

        1. SW_APM_TRANSACTION_NAME
        2. Any instrumentor-set span attributes for FaaS
        3. AWS_LAMBDA_FUNCTION_NAME
        4. Any instrumentor-set span attributes for HTTP
        5. Span name (default)
        6. "unknown" (when limit reached)

        Also caches entry span at its trace ID for custom transaction naming with SDK.
        """
```

jerrytfleung replied:
Done.
Corrected 
6. "other" (when the transaction name pool limit reached)

tammy-baylis-swi commented on solarwinds_apm/trace/serviceentry_processor.py:
```diff
@@ -78,12 +78,15 @@ def on_start(
         http_route = span.attributes.get(SpanAttributes.HTTP_ROUTE, None)
         url_path = span.attributes.get(SpanAttributes.URL_PATH, None)
-        if faas_name:
+        if sw_apm_txn_name:
```
Could we also add a bit of test coverage to `test_serviceentry_processor.py` for this new case

jerrytfleung replied:
Sure. Copied the test case of AWS_LAMBDA_FUNCTION_NAME and built the SW_APM_TRANSACTION_NAME checking.

tammy-baylis-swi commented on tests/unit/test_processors/test_otlp_metrics_processor.py:
```diff
@@ -43,73 +43,73 @@ def test__init(self, mocker):
+    #         mock_apm_config,
+    #     ).calculate_otlp_transaction_name("foo-span")
+    #
+    # def test_calculate_otlp_transaction_name_env_trans_truncated(self, mocker):
```
Oh right these. We're not truncating in ServiceEntryProcessor as we chatted offline yesterday, so we could completely remove these test cases maybe?

jerrytfleung replied:
Removed. Done

tammy-baylis-swi commented:
So sorry, I've introduced a special merge conflict from #568 where I renamed a `test_oltp_metrics_processor.py` to `test_response_time_processor.py`

jerrytfleung commented:
> So sorry, I've introduced a special merge conflict from #568 where I renamed a `test_oltp_metrics_processor.py` to `test_response_time_processor.py`

No problem. Resolved.

tammy-baylis-swi commented on solarwinds_apm/trace/serviceentry_processor.py:
```diff
@@ -65,7 +65,20 @@ def on_start(
+        5. Span name (default)
+        6. "other" (when the transaction name pool limit reached)
+
+        Also caches entry span at its trace ID for custom transaction naming with SDK.
+
+        If entry span, caches it at its trace ID. Used for custom transaction naming.
```
Thanks! Nit: We just need one of these lines

```suggestion
        If entry span, caches it at its trace ID. Used for custom transaction naming.
```

tammy-baylis-swi commented on tests/unit/test_processors/test_response_time_processor.py:
```diff
@@ -343,10 +343,10 @@ def patch_for_on_end(
-            "solarwinds_apm.trace.ResponseTimeProcessor.calculate_otlp_transaction_name",
-            return_value="foo",
-        )
+        # mocker.patch(
+        #     "solarwinds_apm.trace.ResponseTimeProcessor.calculate_otlp_transaction_name",
+        #     return_value="foo",
+        # )
```
Nit: can remove this now too

tammy-baylis-swi reviewed (approved):
Lgtm and thanks for changes. Just nits now.

I also did a manual test looks good. Details in ticket comment.
